### PR TITLE
#1035 Distribution new customer enhancements

### PR DESCRIPTION
**Pull request by Ishaanjolly**
Distribution new customer enhancements

## Description

I added `n_samples: int = 1000` to `_distribution_new_customers` and modified this line within: 
```
        if dataset.sizes["chain"] == 1 and dataset.sizes["draw"] == 1:
            # For map fit add a dummy draw dimension
            dataset = dataset.squeeze("draw").expand_dims(draw=range(n_samples))
```
Additionally:  
I added `n_samples: int = 1` to `distribution_new_customer_recency_frequency` and modified this: 

```
self._distribution_new_customers(
            data=data,
            T=T,
            random_seed=random_seed,
            var_names=["recency_frequency"],
            n_samples=n_samples,
        )["recency_frequency"]
```

## Related Issue

- [ ] Closes #
- [x] Related to #1035 

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [x] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1061.org.readthedocs.build/en/1061/

### CODE REVIEW

File: pymc_marketing/mmm/plot.py [hunk 1]
```diff
@@ -570,7 +570,7 @@ def plot_curve(
     subplot_kwargs : dict, optional
         Addtional kwargs to while creating the fig and axes
     sample_kwargs : dict, optional
-        Kwargs for the :func:`plot_curve` function
+        Kwargs for the :func:`plot_sample` function
```
Comment (juanitorduz) [inline pymc_marketing/mmm/plot.py:573]: Can we leave this change out of this PR? 🙏

**Review by juanitorduz — CHANGES_REQUESTED** [review]

File: pymc_marketing/mmm/plot.py (hunk 1 above)
Comment (Ishaanjolly) [reply pymc_marketing/mmm/plot.py]: Just done this thanks

**Review by juanitorduz — APPROVED** [review]

**Comment by juanitorduz** [comment]
Thank you @Ishaanjolly !

**Comment by ColtAllen** [comment]
Sorry that I was late in seeing this, but the same changes can also be applied to `ParetoNBDModel`. It's also worth noting this PR is related to https://github.com/pymc-labs/pymc-marketing/issues/878 but I'm not entirely sure if that issue is worth pursuing.

**Comment by juanitorduz** [comment]
Thanks @ColtAllen ! @Ishaanjolly would you wanna do something similar to `ParetoNBDModel`?

**Comment by Ishaanjolly** [comment]
Sure, I'll create a new issue for it and pursue it. Thanks @juanitorduz and @ColtAllen 

@ColtAllen  would also appreciate your suggestions on best way to cache log_p outputs.

**Pull request merged**